app/user.go

```diff
@@ -374,16 +361,17 @@ func (a *App) CreateOAuthUser(service string, userData io.Reader, teamID string,
 		}
 	}
 
-	if result := <-suchan; result.NErr == nil {
-		return result.Data.(*model.User), nil
+	userByAuth, _ := a.Srv().Store.User().GetByAuth(user.AuthData, service)
```

Comment:
Let's handle the error

Reply:
Actually, we do handling the error implicitly, if both user fetches are nil then we proceed on creating a user from scratch

Reply:
Ah yes I see now. Perhaps might be good to comment that to help with readability.

Reply:
I don't think this code is exactly equivalent. The old code said "if retrieving the user by service x didn't throw an error then return the user". The new code says "if the returned object is not nil then return the user". One must look into the implementation of `(SqlUserStore).GetByAuth` to know whether a `nil` value is definitely returned in error cases. It wouldn't be unheard of for an instantiated struct to be returned with an error, in which case this logic would break unbeknownst to the developer refactoring `(SqlUserStore).GetByAuth`.